### PR TITLE
docs: Add OpenAPI specification for Curriculum API

### LINKED ISSUE #1: Design Curriculum API
Create API to support the UI that apprentices use to manage their curriculum. 
- [ ] Create OpenApi Specification.
- [ ] Open draft PR and get reviews from UI and Data engineers.

### PR DESCRIPTION
**Pull request by fazzy12**
This commit adds the OpenAPI specification YAML file for the Curriculum API. The YAML file includes endpoints, schemas, and descriptions for managing curriculum resources. It follows the OpenAPI 3.0.3 specification and includes paths for adding, updating, and retrieving curriculum information. 

Closes #1

### CODE REVIEW

File: docs/openapi.yaml [hunk 1]
```diff
@@ -1,88 +1,86 @@
...
             application/json:
               schema:
-                $ref: '#/components/schemas/Person'          
+                $ref: '#/components/schemas/Curriculum'
         '405':
           description: Invalid input
     get:
-      summary: Get a list of people names and ids
-      operationId: getNames
+      summary: Get a list of all curricula
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:34]: Likewise there will be no need to get a list of curriculum, so this endpoint can be dropped as well

File: docs/openapi.yaml [hunk 2]
```diff
@@ -1,88 +1,86 @@
...
-          description: person not found
+          description: Curriculum not found
         '405':
           description: Validation exception
     get:
-      summary: Get an existing person
-      description: Get an existing person by Id
-      operationId: gettPerson
+      summary: Get an existing curriculum
+      operationId: getCurriculum
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:79]: Add a comment here that states if a curriculum does not exist it will be created

File: docs/openapi.yaml [hunk 3]
```diff
@@ -141,164 +106,147 @@ paths:
...
+        _id:
+          description: The unique identifier for a curriculum
+          type: string
+          format: UUID
+        name:
+          description: The name of the curriculum
+          type: string
+        version:
+          description: The version of the curriculum
+          type: string
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:150]: version properties do not be need to be exposed they are going to be deprecated soon

File: docs/openapi.yaml [hunk 4]
```diff
@@ -141,164 +106,147 @@ paths:
...
+          type: array
+          items:
+            $ref: '#/components/schemas/Resource'
+    Resource:
+      type: object
+      required:
+        - type
+        - review
+        - roadmap
+      properties:
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:162]: I don't think review or roadmap need to be required

File: docs/openapi.yaml [hunk 5]
```diff
@@ -141,164 +106,147 @@ paths:
...
+          type: string
+        review:
+          description: Free text review of the person's experience with this resource
+          type: string
+        roadmap:
+          description: The status of the resource in the learning roadmap
+          type: string
+          enum:
+            - completed
+            - celebrated
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:191]: Does this match the schema for the database, I don't think we need both completed and celebrated, they mean the same thing

File: docs/openapi.yaml [hunk 6]
```diff
@@ -141,164 +106,147 @@ paths:
...
+            - now
+            - next
+            - later
+        status:
+          description: Archived, Assigned, Started, Completed
+          type: string
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:197]: The value should be listed as enum

File: docs/openapi.yaml [hunk 7]
```diff
@@ -141,164 +106,147 @@ paths:
...
+          format: date-time
+        rating:
+          description: Rating of the resource (0-5)
+          type: integer
+          minimum: 0
+          maximum: 5
     Config:
       type: object
       properties:
         apiVersion:
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:218]: We will strengthen the config schema as you build the API

**Review by FlatBallFlyer — CHANGES_REQUESTED** [review]
I think we also need to add Post patch delete endpoints for curriculum/ID/Topic. I would also appreciate it if you would put anHTML file in the docs folder similar to the one in the person API so that we can publish this swagger

File: docs/openapi.yaml [hunk 8]
```diff
@@ -1,304 +1,349 @@
...
-                $ref: '#/components/schemas/Person'          
+                $ref: '#/components/schemas/Curriculum'
         '404':
-          description: person not found
+          description: Curriculum not found
         '405':
           description: Validation exception
-  /api/partners/:
+
+  /api/config/:
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:77]: It's a little bit nitpicky but can we move config and health to be the last two in points

File: docs/openapi.yaml [hunk 9]
```diff
@@ -1,304 +1,349 @@
...
-      summary: Get API Configuration Information
-      operationId: getConfig
+                $ref: '#/components/schemas/Topic'
+        '404':
+          description: Curriculum not found
+        '405':
+          description: Validation exception
+
+    patch:
+      summary: Update a topic within a curriculum
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:149]: The patch needs a URL parameter to identify which topic to update

File: docs/openapi.yaml [hunk 10]
```diff
@@ -1,304 +1,349 @@
...
-      summary: Promethius Healthcheck endpoint
-      operationId: getHealth
+                $ref: '#/components/schemas/Topic'
+        '404':
+          description: Topic not found
+        '405':
+          description: Validation exception
+
+    delete:
+      summary: Delete a topic from a curriculum
```
Comment (FlatBallFlyer) [inline docs/openapi.yaml:188]: This delete operation also needs topic ID to identify the topic to delete

**Review by FlatBallFlyer — CHANGES_REQUESTED** [review]
This is looking good, just a few minor changes here, you provided the type for those properties in the schema.

**Comment by fazzy12** [comment]
- Added missing descriptions for 405 responses in the openapi.yaml.
- Rearranged endpoints to position /api/config/ and /api/health/ as the last two points, improving readability.
- Enhanced the patch endpoint to include a URL parameter for specifying the topic to update.
- Modified the delete operation to incorporate a topic ID parameter for precisely identifying the topic to delete."

**Review by FlatBallFlyer — APPROVED** [review]
Draft to be updated when mongo schema is implemented

**Pull request merged**